src/gui/properties/peerlistwidget.cpp

```diff
@@ -364,7 +366,8 @@ void PeerListWidget::loadPeers(const BitTorrent::TorrentHandle *torrent)
 
     // Remove peers that are gone
     for (const PeerEndpoint &peerEndpoint : asConst(existingPeers)) {
-        const QStandardItem *item = m_peerItems.take(peerEndpoint);
+        QStandardItem *item = m_peerItems.take(peerEndpoint);
+        m_addressItems[peerEndpoint.address.ip].remove(item);
```

Comment:
m_addressItems confuses. I would rename it to m_itemsByIP.

Reply:
Shouldn't you also remove entry if it's empty?

Reply:
both are done.